### PR TITLE
Add microseconds persistence support for datetime/time types + fix normalization/cloning issue

### PR DESCRIPTION
**Pull request by mvorisek**
Very important to not loose time precision. All major database engines ignore the microseconds part if the target datatype has less or only integer seconds precision.

### CODE REVIEW

**Comment by mvorisek** [comment]
PR complete, ready to be merged.

File: src/Field.php [hunk 1]
```diff
@@ -409,11 +409,23 @@ public function toString($value = null)
...
+                        if ($this->type === 'date') {
+                            $format = $dateFormat;
+                        } elseif ($this->type === 'time') {
+                            $format = $timeFormat;
+                        } else {
+                            $format = $dateFormat.'\T'.$timeFormat.'P'; // ISO 8601 format 2004-02-12T15:19:21+00:00
+                        }
+
+                        return $v->format($format);
+                    } else {
```
Comment (DarkSide666) [inline src/Field.php]: no need for `else` here - just `return` after if condition :)

File: src/Persistence/SQL.php [hunk 2]
```diff
@@ -533,20 +533,27 @@ public function _typecastLoadField(Field $f, $value)
...
+                    }
+                }
 
                 // datetime only - set from persisting timezone
                 if ($f->type == 'datetime' && isset($f->persist_timezone)) {
                     $v = $dt_class::createFromFormat($format, $v, new $tz_class($f->persist_timezone));
-                    if ($v === false) {
-                        throw new Exception(['Incorrectly formatted datetime', 'format' => $format, 'value' => $value, 'field' => $f]);
+                    if ($v !== false) {
+                        $v->setTimeZone(new $tz_class(date_default_timezone_get()));
```
Comment (DarkSide666) [inline src/Persistence/SQL.php]: should be `setTimezone` instead of `setTimeZone`
https://www.php.net/manual/en/datetime.settimezone.php

Comment (mvorisek) [reply src/Persistence/SQL.php]: Fixed in the another PR https://github.com/atk4/data/pull/505 - merge that PR first.

**Comment by mvorisek** [comment]
PR updated and ready to be merged. Please see comments as another PR should be merged first.

**Pull request merged**